python/ray/data/dataset.py

```diff
@@ -124,6 +124,10 @@ def __init__(
             # TODO(ekl) we should clear inputs once we have full lineage recorded.
             self._plan.execute(clear_input_blocks=False)
 
+    @staticmethod
+    def copy(dataset: "Dataset[T]") -> "Dataset[T]":
+        return Dataset(dataset._plan, dataset._epoch, dataset._lazy)
+
```

Comment:
Nit: I feel like this would be cleaner as an instance method:
```suggestion
    def copy(self) -> "Dataset[T]":
        return Dataset(self._plan, self._epoch, self._lazy)

```